projects/libyaml/libyaml_reformatter_fuzzer.c

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "yaml.h"
+#include <assert.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+#ifdef NDEBUG
+#undef NDEBUG
+#endif
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  if (size < 2) return 0;
+
+  bool done = false;
+  bool is_canonical = data[0] & 1;
+  bool is_unicode = data[1] & 1;
+
+  yaml_parser_t parser;
+  yaml_emitter_t emitter;
+  yaml_event_t event;
+  FILE *f;
+
+  /* Clear the objects. */
+
+  memset(&parser, 0, sizeof(parser));
```

Comment:
are these memset's needed? sometimes e.g. the parser is memset'ed sometimes (in other fuzzers) it is not...